libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshControllerTest.kt

```diff
@@ -449,7 +476,20 @@ class RouteRefreshControllerTest {
             routeRefreshStub.setRefreshedRoute(initialRoute)
             advanceTimeBy(routeRefreshOptions.intervalMillis)
             // assert
-            assertEquals(listOf(initialRoute), refreshedRoute.getCompletedTest())
+            assertEquals(
+                RefreshedRouteInfo(listOf(initialRoute), currentIndicesSnapshot),
+                refreshedRoute.getCompletedTest()
+            )
+            // 304 = 3(attempts) * (100 + 1) + 1
+            coVerify(exactly = 304) {
+                currentIndicesSnapshotProvider.getFilledIndicesAndFreeze()
+            }
+            // 101 = 100 + 1
+            verify(exactly = 101) { currentIndicesSnapshotProvider.invoke() }
+            // 406 = 4(attempts + finally) * (100 + 1) + 1 * 2 (for success)
+            verify(exactly = 406) {
+                currentIndicesSnapshotProvider.unfreeze()
+            }
```

Comment:
Why do we need to check exact number of invocations? The provider should be mocked to return what we want to test and the result is already asserted above. All that we seem to need to check is that by the end of the operation we're not frozen.

Reply:
This is a nit if we're moving away from this approach anyway, so let's leave as is.